### PR TITLE
Use NuGet SteamAuth

### PR DESCRIPTION
**Pull request by BlueRaja**
I created a NuGet package for SteamAuth [here](https://www.nuget.org/packages/SteamAuth/).  This PR updates SteamBot to use it.

This should fix the troubles people have been having with the submodule (eg. #897)

### CODE REVIEW

**Comment by waylaidwanderer** [comment]
It would be better if @geel9 or @Jessecar96 were the ones to do it, but since they haven't...

Looks good to me, except for one thing.

File: SteamBot.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Express 2013 for Windows Desktop
-VisualStudioVersion = 12.0.30501.0
+# Visual Studio 14
+VisualStudioVersion = 14.0.24720.0
```
Comment (waylaidwanderer) [inline SteamBot.sln]: Does this stop users from being able to open the solution in anything older than Visual Studio 14? I remember that being an issue before.

Comment (BlueRaja) [inline SteamBot.sln]: I don't _think_ so, but I don't have VS 2013 to test with.  Can anyone here test it?

**Comment by BlueRaja** [comment]
In the [SteamAuth PR](https://github.com/geel9/SteamAuth/pull/30) I offered to add them both as owners, so they can push new versions when needed.

**Comment by BlueRaja** [comment]
Updated to the latest version of SteamAuth, and reverted the change called out by @waylaidwanderer above.

Has anyone else attempted to test this yet?

**Comment by Bottswana** [comment]
I tried it recently and couldn't get it to work, but it might just be a bad
merge with my tree. I did only look at it for 5 minutes. Will have a
further look tomorrow

**Comment by scholtzm** [comment]
Just merge this. I can't imagine this causing more trouble than the current state.

**Comment by BlueRaja** [comment]
Yeah, that's probably a good idea.  We've had enough "Jsonsoft.dll can't be found" issues...

**Pull request merged**